### PR TITLE
Autotestes: #3289 - Clear Canvas

### PR DESCRIPTION
**Pull request by MartaWilliams**
- 6 testes
- snapshots
- 1 test data file

## How the feature works? / How did you fix the issue?
(Screenshots, videos, or GIFs, if applicable)

## Check list
- [ ] unit-tests written
- [x] e2e-tests written
- [x] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [x] base branch (master or release/xx) is correct
- [x] task status changed to "Code review"
- [x] reviewers are notified about the pull request

### CODE REVIEW

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  LeftPanelButton,
+  RingButton,
+  TopPanelButton,
+  selectRing,
+} from '@utils/selectors';
+import { addTextBoxToCanvas } from '@utils/selectors/addTextBoxToCanvas';
+
+test.describe('Clear canvas', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto('');
```
Comment (rrodionov91) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:21]: i believe we need to use waitForPageInit now

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  });
+  test('Clear Canvas - "Clear canvas" button', async ({ page }) => {
+    // Test case:EPMLSOPKET-1702
+    await selectTopPanelButton(TopPanelButton.Clear, page);
+    await page.getByRole('button', { name: 'Clear Canvas (Ctrl+Del)' }).hover();
+    await delay(DELAY_IN_SECONDS.TWO);
+    await page.getByTitle('Clear Canvas (Ctrl+Del)').click();
```
Comment (rrodionov91) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:28]: Why do we need this part?

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  test('Clear Canvas - Structure is opened from smile-string', async ({
+    page,
+  }) => {
+    // Test case: EPMLSOPKET-1706
+    await openFileAndAddToCanvas('chain-with-r-group.smi', page);
+    await clickInTheMiddleOfTheScreen(page);
+    await selectTopPanelButton(TopPanelButton.Clear, page);
+    await selectTopPanelButton(TopPanelButton.Undo, page);
+    await page.keyboard.press('Control+Delete');
+    await delay(DELAY_IN_SECONDS.TWO);
```
Comment (rrodionov91) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:83]: Why do we need delay here?

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    await selectTopPanelButton(TopPanelButton.Undo, page);
+    await selectTopPanelButton(TopPanelButton.Undo, page);
+    await selectTopPanelButton(TopPanelButton.Redo, page);
+    await selectTopPanelButton(TopPanelButton.Redo, page);
+    await takeEditorScreenshot(page);
+  });
+  test('Clear Canvas - Pressing Clear Canvas with Reaction Arrow under mouse cursor not causes errors in DevTool console', async ({
+    page,
+  }) => {
+    // await page.keyboard.press('Control+Shift+j');
```
Comment (rrodionov91) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:93]: Why is it commented?
Also There is no devtools on screenshot

**Review by rrodionov91 — CHANGES_REQUESTED** [review]

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts (hunk 4 above)
Comment (MartaWilliams) [reply ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:93]: Mikhail told me to leave this one, as we don’t yet have a function to track errors in the console

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts (hunk 2 above)
Comment (MartaWilliams) [reply ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:28]: Steps in the test: 
1. Move the cursor over the button.
2. Click the 'Clear Canvas' button.

**Review by AnastasiyaPiatrovaKlu — APPROVED** [review]

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  test.beforeEach(async ({ page }) => {
+    await waitForPageInit(page);
+  });
+
+  test('Clear Canvas - "Clear canvas" button', async ({ page }) => {
+    // Test case:EPMLSOPKET-1702
+    // Checking apperance of the tooltip and deleting object with Clear canvas button
+    // await selectTopPanelButton(TopPanelButton.Clear, page);
+    await page.getByRole('button', { name: 'Clear Canvas (Ctrl+Del)' }).hover();
+    await page.getByTitle('Clear Canvas (Ctrl+Del)').click();
```
Comment (Nitvex) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:24]: You don't need to hover the mouse over the button. We already have a function to clear on Clear canvas button

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    await page.getByRole('dialog').getByRole('textbox').fill('12345');
+    await pressButton(page, 'Apply');
+    await selectTopPanelButton(TopPanelButton.Clear, page);
+    await takeEditorScreenshot(page);
+  });
+
+  test('Clear Canvas - Undo/Redo', async ({ page }) => {
+    // Test case: EPMLSOPKET-1704
+    // Checking deleting object with Clear canvas button with funcion of Undo/Redo buttons
+    await openFileAndAddToCanvas('reaction-dif-prop.rxn', page);
```
Comment (Nitvex) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:35]: Could you, please, put the file to corresponding folder?

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+  });
+
+  test('Clear Canvas - "Clear canvas" button', async ({ page }) => {
+    // Test case:EPMLSOPKET-1702
+    // Checking apperance of the tooltip and deleting object with Clear canvas button
+    // await selectTopPanelButton(TopPanelButton.Clear, page);
+    await page.getByRole('button', { name: 'Clear Canvas (Ctrl+Del)' }).hover();
+    await page.getByTitle('Clear Canvas (Ctrl+Del)').click();
+    await addTextBoxToCanvas(page);
+    await page.getByRole('dialog').getByRole('textbox').fill('12345');
```
Comment (Nitvex) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:26]: Please, do not use magic numbers

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    await takeEditorScreenshot(page);
+    await page.keyboard.press('Control+Delete');
+    await takeEditorScreenshot(page);
+  });
+
+  test('Clear Canvas - Structure is opened from smile-string', async ({
+    page,
+  }) => {
+    // Test case: EPMLSOPKET-1706
+    // Checking deleting object with Clear canvas button with funcion of hotkyes (Ctrl+ Del) andf Undo/Redo button
```
Comment (Nitvex) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:81]: A typo "andf"

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+    await selectTopPanelButton(TopPanelButton.Undo, page);
+    await selectTopPanelButton(TopPanelButton.Redo, page);
+    await selectTopPanelButton(TopPanelButton.Undo, page);
+    await takeEditorScreenshot(page);
+  });
+
+  test('Clear Canvas - Structure is opened from ket-file', async ({ page }) => {
+    // Test case:EPMLSOPKET-1705
+    // Checking deleting object with Clear canvas button with funcion of hotkyes (Ctrl+ Del)
+    await openFileAndAddToCanvas('ketcher.ket', page);
```
Comment (Nitvex) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:49]: Why do you perform multiple actions in test? You write
"Checking deleting object with Clear canvas button with funcion of hotkyes (Ctrl+ Del)"
What do you check? Do you check that canvas is cleared after clicking the button in top panel? Or do you check that canvas is cleared after shortcut?
If you want to check both, then you need to create 2 separate tests

**Review by chgayane — APPROVED** [review]

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+  TopPanelButton,
+  takeEditorScreenshot,
+  selectTopPanelButton,
+  waitForPageInit,
+  clickInTheMiddleOfTheScreen,
+  openFileAndAddToCanvas,
+  selectRing,
+  RingButton,
+} from '@utils';
+// import { addTextBoxToCanvas } from '@utils/selectors/addTextBoxToCanvas';
```
Comment (Nitvex) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:12]: Please, remove the comment

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+  test.afterEach(async ({ page }) => {
+    await takeEditorScreenshot(page);
+  });
+
+  test('Clear Canvas - "Clear canvas" button', async ({ page }) => {
+    // Test case:EPMLSOPKET-1702
+    // Checking apperance of the tooltip and deleting object with Clear canvas button
+    await selectTopPanelButton(TopPanelButton.Clear, page);
+    await MouseEvent;
```
Comment (Nitvex) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:27]: Why have you added this?

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+    await selectTopPanelButton(TopPanelButton.Undo, page);
+    await page.keyboard.press('Control+Delete');
+    await selectTopPanelButton(TopPanelButton.Undo, page);
+    await selectTopPanelButton(TopPanelButton.Undo, page);
+    await selectTopPanelButton(TopPanelButton.Redo, page);
+    await selectTopPanelButton(TopPanelButton.Redo, page);
+    await takeEditorScreenshot(page);
+  });
+
+  // test('Clear Canvas - Pressing Clear Canvas with Reaction Arrow under mouse cursor not causes errors in DevTool console', async ({
```
Comment (Nitvex) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:93]: Please, remove this comment

**Review by Nitvex — CHANGES_REQUESTED** [review]

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+  selectTopPanelButton,
+  waitForPageInit,
+  waitForRender,
+} from '@utils';
+
+async function addTextBoxToCanvas(page: Page) {
+  await page.getByTitle('Add text (Alt+T)').click();
+  await clickInTheMiddleOfTheScreen(page);
+  await page.getByRole('dialog').getByRole('textbox').click();
+}
```
Comment (rrodionov91) [inline ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:19]: I believe it's better to move this function to utils to reuse it it other tests

**Review by rrodionov91 — CHANGES_REQUESTED** [review]

File: ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts (hunk 13 above)
Comment (MartaWilliams) [reply ketcher-autotests/tests/User-Interface/Clear-Canvas/clear-canvas.spec.ts:19]: Done! :)

**Review by rrodionov91 — APPROVED** [review]

**Review by chgayane — APPROVED** [review]

**Pull request merged**